src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -292,15 +294,19 @@ public void ExecuteSuppressionAction(DiagnosticSuppressor suppressor, ImmutableA
             cancellationToken.ThrowIfCancellationRequested();
 
             var supportedSuppressions = _analyzerManager.GetSupportedSuppressionDescriptors(suppressor, this, cancellationToken);
-            Func<SuppressionDescriptor, bool> isSupportedSuppression = supportedSuppressions.Contains;
-            Action<SuppressionAnalysisContext> action = suppressor.ReportSuppressions;
```

Comment:
this delegate allocation was unnecessary.  we can just pass the suppressor to ExecuteAndCatchIfThrows which can then call .ReportSuppressions on htat without needing a delegate indirectino.